lib/ansible/modules/network/nuage/nuage_vspk.py

```diff
@@ -0,0 +1,1089 @@
+    returned: On state=present and command=find in case one entity was found.
+    type: string
+    sample: bae07d8d-d29c-4e2b-b6ba-621b4807a333
+entities:
+    description: A list of entities handled. Each element is the to_dict() of the entity.
+    returned: On state=present and find, with only one element in case of state=present or find=one.
+    type: list
+    sample: [{
+        "ID": acabc435-3946-4117-a719-b8895a335830",
+        "assocEntityType": "DOMAIN",
+        "command": "BEGIN_POLICY_CHANGES",
+        "creationDate": 1487515656000,
+        "entityScope": "ENTERPRISE",
+        "externalID": null,
+        "lastUpdatedBy": "8a6f0e20-a4db-4878-ad84-9cc61756cd5e",
+        "lastUpdatedDate": 1487515656000,
+        "owner": "8a6f0e20-a4db-4878-ad84-9cc61756cd5e",
+        "parameters": null,
+        "parentID": "a22fddb9-3da4-4945-bd2e-9d27fe3d62e0",
+        "parentType": "domain",
+        "progress": 0.0,
+        "result": null,
+        "status": "RUNNING"
+        }]
+'''
+
+import time
+import importlib
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from bambou.exceptions import BambouHTTPError
+    HASBAMBOU = True
+except ImportError:
+    HASBAMBOU = False
+
+SUPPORTED_COMMANDS = ['find', 'change_password', 'wait_for_job', 'get_csp_enterprise']
+VSPK = None
+
+
+class NuageEntityManager(object):
+    """
+    This module is meant to manage an entity in a Nuage VSP Platform
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.auth = module.params['auth']
+        self.api_username = None
+        self.api_password = None
+        self.api_enterprise = None
+        self.api_url = None
+        self.api_version = None
+        self.api_certificate = None
+        self.api_key = None
+        self.type = module.params['type']
+
+        self.state = None
+        if 'state' in list(module.params.keys()):
+            self.state = module.params['state']
+
+        self.command = None
+        if 'command' in list(module.params.keys()):
+            self.command = module.params['command']
+
+        self.match_filter = None
+        if 'match_filter' in list(module.params.keys()):
+            self.match_filter = module.params['match_filter']
+
+        self.entity_id = None
+        if 'id' in list(module.params.keys()):
+            self.entity_id = module.params['id']
+
+        self.parent_id = None
+        if 'parent_id' in list(module.params.keys()):
+            self.parent_id = module.params['parent_id']
+
+        self.parent_type = None
+        if 'parent_type' in list(module.params.keys()):
+            self.parent_type = module.params['parent_type']
+
+        self.properties = None
+        if 'properties' in list(module.params.keys()):
+            self.properties = module.params['properties']
+
+        self.children = None
+        if 'children' in list(module.params.keys()):
```

Comment:
Since `children`, `properties` and other's are defined as module args, the key will be present in `params` and initialized to is a default value. Hence the `if` check here and above is not required.
`self.children = module.params['children']` is sufficient